src/main/kotlin/fi/metatavu/timebank/api/impl/VacationsApi.kt

```diff
@@ -0,0 +1,54 @@
+package fi.metatavu.timebank.api.impl
+
+import fi.metatavu.timebank.api.controllers.VacationController
+import fi.metatavu.timebank.api.impl.translate.VacationRequestTranslator
+import fi.metatavu.timebank.model.VacationRequest
+import javax.enterprise.context.RequestScoped
+import fi.metatavu.timebank.spec.VacationsApi
+import java.time.LocalDate
+import java.util.*
+import javax.inject.Inject
+import javax.ws.rs.core.Response
+
+/**
+ * API implementation for TimeEntries API
+ */
+@RequestScoped
+class VacationsApi: VacationsApi, AbstractApi() {
+
+    @Inject
+    lateinit var vacationController: VacationController
+
+    @Inject
+    lateinit var vacationRequestTranslator: VacationRequestTranslator
+
+    override suspend fun deleteVacationRequest(id: UUID): Response {
+        loggedUserId ?: return createUnauthorized(message = "Invalid token!")
+
+        vacationController.deleteRequest(id = id)
+
+        return createNoContent()
+    }
+
+    override suspend fun createVacationRequest(vacationRequest: VacationRequest): Response {
+        loggedUserId ?: return createUnauthorized(message = "Invalid token!")
+
+        vacationController.createVacationRequest(vacationRequest = vacationRequest)
```

Comment:
Again could this be (vacationRequest) rather than (vacationRequest = vacationRequest)